src/avlizer_confluent.erl

```diff
@@ -353,21 +366,32 @@ unify_ref(Ref) -> Ref.
 
 -spec do_download(string(), ref()) -> {ok, binary()} | {error, any()}.
 do_download(RegistryURL, RegId) when is_integer(RegId) ->
+  RegIdBin = integer_to_binary(RegId),
   URL = RegistryURL ++ "/schemas/ids/" ++ integer_to_list(RegId),
-  httpc_download(URL);
+   case httpc_download(URL) of
+     {ok, RegIdBin, Schema} ->
+       {ok, Schema};
+     Error ->
+       Error
+   end;
 do_download(RegistryURL, {Name, Fp}) ->
   Subject = fp_to_subject(Name, Fp),
   %% fingerprint is unique, hence always version/1
   URL = RegistryURL ++ "/subjects/" ++ Subject ++ "/versions/1",
-  httpc_download(URL).
+  case httpc_download(URL) of
+    {ok, _RegIdBin, Schema} ->
+      {ok, Schema};
+    Error ->
+      Error
+  end.
 
 httpc_download(URL) ->
   case httpc:request(get, {URL, _Headers = []},
                      [{timeout, ?HTTPC_TIMEOUT}], []) of
     {ok, {{_, OK, _}, _RspHeaders, RspBody}} when OK >= 200, OK < 300 ->
-      #{<<"schema">> := SchemaJSON} =
+      #{<<"schema">> := SchemaJSON, <<"id">> := Id} =
```

Comment:
`httpc_download` function is currently used in two places:
1) `"/schemas/ids/" ++ integer_to_list(RegId)`
2) `"/subjects/" ++ Subject ++ "/versions/1"`

The former URL does indeed include "id" field in the response: https://docs.confluent.io/current/schema-registry/develop/api.html#get--subjects-(string-%20subject)-versions-(versionId-%20version) ...But the latter doesn't: https://docs.confluent.io/current/schema-registry/develop/api.html#get--schemas-ids-int-%20id

It looks like this change may break fetching schema by id.